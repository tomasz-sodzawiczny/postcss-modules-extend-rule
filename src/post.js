import chainPlugins from "postcss-plugin-chain";
import extendRule from "postcss-extend-rule";
import extendClass from "./plugins/postcss-modules-extend-class";

const plugin = chainPlugins("postcss-modules-extend-rule/pre", [
  extendClass,
  extendRule
]);

export default plugin;
