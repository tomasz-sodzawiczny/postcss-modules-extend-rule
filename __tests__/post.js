import post from "../src/post";

const process = css =>
  post.process(css, { from: undefined }).then(result => result.css);

describe("postcss-modules-extends-rule/post", () => {
  it("works", async () => {
    const css =
      ".blue {color: blue} .a { @postcss-modules-extend-class blue; }";

    expect(await process(css)).toMatchSnapshot();
  });
});
