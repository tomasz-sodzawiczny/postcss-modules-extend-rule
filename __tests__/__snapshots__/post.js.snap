// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-modules-extends-rule/post works 1`] = `".blue {color: blue} .a {color: blue}"`;
